src/fnargs.c

```diff
@@ -136,7 +159,13 @@ int prepare_fn_args_specs(int func_id, const struct mass_attacher_func_info *fin
 	struct func_arg_spec *spec;
 	const struct btf_type *fn_t, *t;
 	int i, n, reg_idx = 0;
+#if defined(__x86_64__)
 	int stack_off = 8; /* 8 bytes for return address */
+#elif defined(__aarch64__)
+	int stack_off = 0; /* ARM64 uses x0-x7 for first 8 args */
```

Comment:
not sure what this comment has to do with stack offset... first few args will be in registers, but the rest might be on the stack, just like for x86-64

Reply:
addressed in https://github.com/anakryiko/retsnoop/pull/89/commits/0aacc5864af055e4a36834d8c914427fb7f26295